### ISSUE TITLE
S3996 should only be reporting identifiers that end with Url.

### ISSUE THREAD

**Issue by StingyJack** [SonarSource/sonar-dotnet#4045]
### Description

a property like 
``` csharp
        public string SocialAuthConfigUrlKeyValue { get; set; }
```
triggers this when its clearly going to be a KeyValue stored in the property and not a Url.

### Repro steps

see description

### Expected behavior

I expect no flagging of this as a violation

### Actual behavior

its an error for me.

### Known workarounds

suppress with an attribute and justification of "False Positive" or "Incorrectly applied sonar rule"

**Comment by pavel-mikula-sonarsource** [comment]
Hi @StingyJack, 

as it's False Positive in this case, the rule still makes sense as it's implemented. Checking only suffix would not match cases like `UrlLogin` or a method `CreateUrlLogin`

**Comment by StingyJack** [comment]
@pavel-mikula-sonarsource -  In English (the only language I know fluently), adjectives precede a noun (i'm sure there are probably pathological corner cases). This is opposite of every other human language I've been exposed to. 

For example, "The fuzzy dog" would express the idea that a dog had a fuzzy coat. 

Inverting the words like your cited example would be "The dog fuzzy". This would be referring to a dog named Fuzzy, and does not describe the dogs coat at all. _Inverting the adjective and noun is wholly different idea to communicate._ 

Those cases you cited use English words, so English is the language to evaluate with so the noun in those is "Login", not "Url". The variable type checking is evaluating incorrectly and throwing False Positive in all cases where English is the name and the identifier does not end with one of the Url variants.

**Comment by pavel-mikula-sonarsource** [comment]
Our product is not limited to English speaking developers. There are many other languages used to name the members.

**Comment by StingyJack** [comment]
I didn't say that you or anyone had to use English, _you used_ English in your examples, and those examples are grammatically and idiomatically incorrect.   And the program is giving me grief and requiring me to do more work to add a suppression and justification when it should not.

Other languages won't use only English words, and if they do there is little chance that they will use a different grammar, so it's safe to treat them as English grammar.

What is a universal resource identifier known as in other languages? Are Cyrillic and other non latin alphabets covered in your tests?

**Comment by StingyJack** [comment]
Also having this trigger on `urlScheme` which is not a url
